server/src/main.c

```diff
@@ -6,11 +6,128 @@
 ** You can even have multiple lines if you want !
 */
 
+#include <string.h>
+#include <time.h>
 #include "server.h"
+#include "team.h"
+#include "array.h"
+
+static size_t *find_map_size(int ac, char **av)
+{
+    int i = 1;
+    size_t *map_size;
+
+    for (; i < ac && strcmp(av[i], "-x") != 0; i++);
+    if (i >= ac)
+        return NULL;
+    map_size = malloc(sizeof(size_t) * 2);
+    map_size[0] = atoi(av[i + 1]);
+    for (i = 1; i < ac && strcmp(av[i], "-y") != 0; i++);
```

Comment:
This shall be a "while" loop, for loop shall be reserved for times when you know the number of iterations and here is clearly not the case